### PR TITLE
Add chromium integration doc

### PR DESCRIPTION
**Pull request by alexzhaobin**
https://trello.com/c/LI3Vd3sQ/1098-samsung-si-browser-integration-poc-will-need-to-do-a-demo-chromium-integration-again-we-can-send-this-through-asap

### CODE REVIEW

File: docs/sdk/browser/v2/ios/chromium.md [hunk 1]
```diff
@@ -0,0 +1,401 @@
+---
+layout: square
+title: Fillr Android Embedded SDK Chromium Integration Guide
+excerpt: Full instructions for integrating the Fillr Embedded SDK for Chromium Browser.
+download_link: https://fillr.s3.amazonaws.com/sdk/android/v2/fillrsdk_chromium.zip
```
Comment (3000) [inline docs/sdk/browser/v2/ios/chromium.md]: This file is not available!

Also i think we should point them to the Bintray download links so everything is released in the same place?

Comment (3000) [reply docs/sdk/browser/v2/ios/chromium.md]: I assume there is no reason we can't release the updated SDK to bintray right?

File: docs/sdk/browser/v2/ios/chromium.md [hunk 2]
```diff
@@ -0,0 +1,401 @@
...
+
+int getContentViewVisibility();
+//End
+```
+
+10. Add new methods to class /chromium/src/content/public/android/java/src/org/chromium/content/browser/ContentViewCoreImpl.java
+```java
+// Fillr Methods
+public void setAllowJavascriptInterfacesInspection(boolean allow) {
+mWebContents.setAllowJavascriptInterfacesInspection(allow);
```
Comment (3000) [inline docs/sdk/browser/v2/ios/chromium.md]: Indent the code in all these methods!

File: docs/sdk/browser/v2/ios/chromium.md [hunk 3]
```diff
@@ -0,0 +1,401 @@
...
+public class FillrChromiumJSNativeInterface {
+    @JavascriptInterface
+    public void setFields(final String json) {
+        try {
+            Method m = mWidgetInterface.getClass().getMethod("setFields", String.class);
+            m.invoke(mWidgetInterface, json);
+        } catch (Exception e) {
+            Log.e(TAG, "Failed to setFields");
+            e.printStackTrace();
+            }
```
Comment (3000) [inline docs/sdk/browser/v2/ios/chromium.md]: Format?

File: docs/sdk/browser/v2/ios/chromium.md [hunk 4]
```diff
@@ -0,0 +1,401 @@
...
+}
+
+public void setJavaScriptEnabled(boolean isEnabled){
+
+}
+
+boolean addedAndroidInterface = false;
+public void addJavascriptInterface(Object widgetInterface, String jniIdentifier){
+    if (mContentViewCore != null && jniIdentifier == "androidInterface" && !addedAndroidInterface) {
+        this.mWidgetInterface = widgetInterface;
```
Comment (3000) [inline docs/sdk/browser/v2/ios/chromium.md]: Where is the instance variable declared?

Comment (3000) [reply docs/sdk/browser/v2/ios/chromium.md]: Maybe we actually hold this variable in the FillrChromiumJSNativeInterface ?

File: docs/sdk/browser/v2/ios/chromium.md [hunk 5]
```diff
@@ -0,0 +1,401 @@
...
+        } catch (Exception e) {
+            Log.e(TAG, "Failed to setFields");
+            e.printStackTrace();
+        }
+    }
+}
+// End Fillr Interface Methods
+```
+
+9. Add new methods to Interface /chromium/src/content/public/android/java/src/org/chromium/content/browser/ContentViewCore.java
```
Comment (hades-6) [inline docs/sdk/browser/v2/ios/chromium.md]: it says interface, but there's no declaration?

File: docs/sdk/browser/v2/ios/chromium.md [hunk 6]
```diff
@@ -0,0 +1,401 @@
...
+    }
+
+    return View.GONE;
+}
+
+public String getFillrUrl(){
+    return getUrl();
+}
+
+public class FillrChromiumJSNativeInterface {
```
Comment (3000) [inline docs/sdk/browser/v2/ios/chromium.md]: What was the main reason we needed to redefine the interface?

File: docs/sdk/browser/v2/ios/chromium.md [hunk 7]
```diff
@@ -0,0 +1,401 @@
...
+        android:layout_height="50dp"
+        android:layout_gravity="bottom"
+        android:visibility="invisible"/>
+
+</FrameLayout>
+```
+
+12. Update /chromium/src/content/browser/frame_host/render_frame_host_impl.cc at line 192 change
+```xml
+bool g_allow_injecting_javascript = true;
```
Comment (hades-6) [inline docs/sdk/browser/v2/ios/chromium.md]: is this overriding a global flag? is this a potential security flaw? (sorry if this sounds like a dumb q but just want to confirm)

File: docs/sdk/browser/v2/ios/chromium.md [hunk 8]
```diff
@@ -0,0 +1,401 @@
...
+layout: square
+title: Fillr Android Embedded SDK Chromium Integration Guide
+excerpt: Full instructions for integrating the Fillr Embedded SDK for Chromium Browser.
+download_link: https://fillr.s3.amazonaws.com/sdk/android/v2/fillrsdk_chromium.zip
+download_text: Download the Android SDK
+---
+
+### Overview
+
+The Browser Embedded SDK provides an easy way to integrate Fillr autofill in to your own WebView based application, allowing users to populate web forms in your browser. This Integration guide will walk you through adding Fillr autofill to Chromium based project. Please note the integration is based on latest Chromium project which has support of android .aar library files.
```
Comment (running-codebase) [inline docs/sdk/browser/v2/ios/chromium.md]: `autofill in to your` should be `autofill into your`

Comment (running-codebase) [reply docs/sdk/browser/v2/ios/chromium.md]: For `latest Chromium project` could we use a version number here perhaps?

File: docs/sdk/browser/v2/ios/chromium.md [hunk 9]
```diff
@@ -0,0 +1,401 @@
...
+Add following at the end of preInflationStartup method to initialize Fillr SDK:
+```java
+FillrApplication init = FillrApplication.getPopApplication(this);
+init.init();
+
+Fillr fillr = Fillr.getInstance();
+FillrBrowserProperties fillrBrowserProperties = new FillrBrowserProperties("Chromium", "Chromium");
+fillr.initialise("e78dfs9f6asdf2jsda3", "", this, BROWSER_TYPE.WEB_KIT, fillrBrowserProperties);
+```
+Add folowing at the end of onResumeWithNative method:
```
Comment (running-codebase) [inline docs/sdk/browser/v2/ios/chromium.md]: `folowing` should be `following`

File: docs/sdk/browser/v2/ios/chromium.md [hunk 10]
```diff
@@ -0,0 +1,401 @@
...
+```java
+Fillr.getInstance().onResume();
+```
+Add folowing at the end of onActivityResultWithNative method:
```
Comment (running-codebase) [inline docs/sdk/browser/v2/ios/chromium.md]: `folowing` should be `following`

File: docs/sdk/browser/v2/ios/chromium.md [hunk 11]
```diff
@@ -0,0 +1,401 @@
...
+            }
+            }
+
+    @JavascriptInterface
+    public void setFillResult(final String json) {
+        try {
+            Method m = mWidgetInterface.getClass().getMethod("setFillResult", String.class);
+            m.invoke(mWidgetInterface, json);
+        } catch (Exception e) {
+            Log.e(TAG, "Failed to setFields");
```
Comment (running-codebase) [inline docs/sdk/browser/v2/ios/chromium.md]: Should be changed to `Log.e(TAG, "Failed to setFillResult");`

File: docs/sdk/browser/v2/ios/chromium.md [hunk 12]
```diff
@@ -0,0 +1,401 @@
...
+        }
+    }
+
+    @JavascriptInterface
+    public void checkFields(final String json) {
+        try {
+            Method m = mWidgetInterface.getClass().getMethod("checkFields", String.class);
+            m.invoke(mWidgetInterface, json);
+        } catch (Exception e) {
+            Log.e(TAG, "Failed to setFields");
```
Comment (running-codebase) [inline docs/sdk/browser/v2/ios/chromium.md]: Should be changed to `Log.e(TAG, "Failed to checkFields");`

File: docs/sdk/browser/v2/ios/chromium.md [hunk 13]
```diff
@@ -0,0 +1,401 @@
...
+        }
+    }
+
+    @JavascriptInterface
+    public void fieldFocussed(final String json) {
+        try {
+            Method m = mWidgetInterface.getClass().getMethod("fieldFocussed", String.class);
+            m.invoke(mWidgetInterface, json);
+        } catch (Exception e) {
+            Log.e(TAG, "Failed to setFields");
```
Comment (running-codebase) [inline docs/sdk/browser/v2/ios/chromium.md]: Should be changed to `Log.e(TAG, "Failed fieldFocussed");`

File: docs/sdk/browser/v2/ios/chromium.md [hunk 14]
```diff
@@ -0,0 +1,401 @@
...
+        }
+    }
+
+    @JavascriptInterface
+    public void fieldBlurred(final String json) {
+        try {
+            Method m = mWidgetInterface.getClass().getMethod("fieldBlurred", String.class);
+            m.invoke(mWidgetInterface, json);
+        } catch (Exception e) {
+            Log.e(TAG, "Failed to setFields");
```
Comment (running-codebase) [inline docs/sdk/browser/v2/ios/chromium.md]: Should be changed to `Log.e(TAG, "Failed fieldBlurred");`

File: docs/sdk/browser/v2/ios/chromium.md [hunk 15]
```diff
@@ -0,0 +1,401 @@
...
+        }
+    }
+
+    @JavascriptInterface
+    public void setWidgetVersion(final String version) {
+        try {
+            Method m = mWidgetInterface.getClass().getMethod("setWidgetVersion", String.class);
+            m.invoke(mWidgetInterface, version);
+        } catch (Exception e) {
+            Log.e(TAG, "Failed to setFields");
```
Comment (running-codebase) [inline docs/sdk/browser/v2/ios/chromium.md]: Should be changed to Log.e(TAG, "Failed setWidgetVersion");

File: docs/sdk/browser/v2/ios/chromium.md [hunk 16]
```diff
@@ -0,0 +1,401 @@
...
+        }
+    }
+
+    @JavascriptInterface
+    public void formMutation(final String formData) {
+        try {
+            Method m = mWidgetInterface.getClass().getMethod("formMutation", String.class);
+            m.invoke(mWidgetInterface, formData);
+        } catch (Exception e) {
+            Log.e(TAG, "Failed to setFields");
```
Comment (running-codebase) [inline docs/sdk/browser/v2/ios/chromium.md]: Should be changed to Log.e(TAG, "Failed formMutation");

File: docs/sdk/browser/v2/ios/chromium.md [hunk 17]
```diff
@@ -0,0 +1,401 @@
...
+mWebContents.addPossiblyUnsafeJavascriptInterface(object, name, JavascriptInterface.class);
+}
+
+public void evaluateJavaScript(String script, JavaScriptCallback callback) {
+mWebContents.evaluateJavaScript(script, callback);
+}
+
+public boolean fillrHasFocus() {
+if (!mContainerView.isFocusable())
+return true;
```
Comment (running-codebase) [inline docs/sdk/browser/v2/ios/chromium.md]: I'm partial to brackets on all if statements but it is your choice.

File: docs/sdk/browser/v2/ios/chromium.md (hunk 1 above)
Comment (alexzhaobin) [reply docs/sdk/browser/v2/ios/chromium.md]: I can give you that file. I have a little bit worry to use Bintray downloads as complication of this project, there will be more glitches than you can imagine if they download from bintray, the easiest way is just shoot them a folder including everything.

File: docs/sdk/browser/v2/ios/chromium.md (hunk 4 above)
Comment (alexzhaobin) [reply docs/sdk/browser/v2/ios/chromium.md]: Good idea

File: docs/sdk/browser/v2/ios/chromium.md (hunk 5 above)
Comment (alexzhaobin) [reply docs/sdk/browser/v2/ios/chromium.md]: Implementation is in ContentViewCoreImpl

File: docs/sdk/browser/v2/ios/chromium.md (hunk 7 above)
Comment (alexzhaobin) [reply docs/sdk/browser/v2/ios/chromium.md]: It is not a global variable, only used within that file, but don't know why it is named starting with g_

File: docs/sdk/browser/v2/ios/chromium.md (hunk 6 above)
Comment (alexzhaobin) [reply docs/sdk/browser/v2/ios/chromium.md]: Because ours is not compatible with Chromium anymore

**Pull request merged**